distributed/worker.py

```diff
@@ -125,6 +127,51 @@
 
 SerializedTask = namedtuple("SerializedTask", ["function", "args", "kwargs", "task"])
 
+_current_worker: contextvars.ContextVar[Worker] = contextvars.ContextVar(
+    "current_worker"
+)
+
+P = ParamSpec("P")
```

Comment:
I was recently burned by `ParamSpec` in that not all ecosystem libraries supported `typing_extensions~=3.10` (in my case it was tensorflow). That may have improved in the last few months, but something to consider

Reply:
Yeah, trying to get `ParamSpec` working probably took 4x longer than actually implementing the content of this PR. I'm just really excited about it.

Since mypy doesn't even support it properly, there's not much value in it here yet. I had thought we could leave it around and remove the `# type: ignore`s in the future, but I probably should just take it out before merging this.